src/reconcile/rec_write.c

```diff
@@ -1981,10 +2001,12 @@ __rec_split(WT_SESSION_IMPL *session, WT_RECONCILE *r, size_t next_len)
 		 * add whatever space remains in this page.
 		 */
 		len = WT_PTRDIFF32(r->first_free, dsk);
-		if (len + r->split_size <= r->page_size)
+		if (len + r->split_size <= r->page_size) {
 			r->space_avail =
 			    r->split_size - WT_PAGE_HEADER_BYTE_SIZE(btree);
-		else {
+			return (0);
```

Comment:
@agorrod, I'm pretty sure this return is dangerous, I don't think we've confirmed the next key/value pair will fit into the buffer, for example, if the next key/value pair is larger than `r->split_size`.